ValheimPlus/GameClasses/Inventory.cs

```
@@ -108,6 +108,96 @@ private static void Prefix(ref Inventory __instance, ref Inventory fromInventory
         }
     }
 
+    public static class AutoStackAllStore
+    {
+        public static bool isStacking = false;
+        public static Inventory currentInventory = null;
+        public static int lastPlayerItemCount = 0;
+    }
+
+    [HarmonyPatch(typeof(Inventory), nameof(Inventory.StackAll))]
+    public static class Container_StackAll_Patch
+    {
+        /// <summary>
+        /// Call StackAll on all chests in range.
+        /// </summary>
+        static void Postfix(Inventory __instance, ref int __result)
+        {
+            if (!Configuration.Current.Inventory.autoStackAll || (AutoStackAllStore.isStacking && __instance != AutoStackAllStore.currentInventory)) return;
+
+            // get chests in range
+            GameObject pos = Player.m_localPlayer.gameObject;
+            List<Container> chests = InventoryAssistant.GetNearbyChests(pos, Helper.Clamp(Configuration.Current.Inventory.autoStackAllRange, 1, 50), !Configuration.Current.Inventory.autoStackAllIgnorePrivateAreaCheck);
+
+            // try to stack all items on found containers
+            try
+            {
+                foreach (Container container in chests)
+                {
+                    if (container.m_inventory == __instance) continue;
+                    container.StackAll();
+                }
+            }
+            catch (Exception) { }
```

Why swallow any exception? We should find out why there are any instead of stopping once we receive one.

---

Just had that in case it triggers if chests are blocked. But it does not ^^'